src/main/java/de/tum/in/www1/artemis/web/rest/BonusResource.java

```diff
@@ -0,0 +1,253 @@
+        if (bonusTo != null) {
+            // This line breaks the circular dependency between savedBonus and bonusToGradingScale
+            bonusTo.setBonusFrom(null);
+            bonusTo.setGradeSteps(null);
+        }
+
+        GradingScale source = bonus.getSourceGradingScale();
+        if (source != null) {
+            source.setGradeSteps(null);
+        }
+    }
+
+    /**
+     * PUT /courses/{courseId}/bonus : Update bonus source for course
+     *
+     * @param bonus the bonus source which will be updated
+     * @return ResponseEntity with status 200 (Ok) with body the newly updated bonus source if it is correctly formatted and 400 (Bad request) otherwise
+     */
+    @PutMapping("/bonus")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<Bonus> updateBonus(@RequestBody Bonus bonus) {
+        log.debug("REST request to update a bonus source: {}", bonus.getId());
+        Bonus oldBonus = bonusRepository.findById(bonus.getId()).orElseThrow(() -> new EntityNotFoundException("Bonus", bonus.getId()));
+        GradingScale bonusToGradingScale = gradingScaleRepository.findWithEagerBonusFromByBonusFromId(oldBonus.getId())
+                .orElseThrow(() -> new EntityNotFoundException("Grading Scale From Bonus", bonus.getId()));
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, bonusToGradingScale.getExam().getCourse(), null);
+
+        boolean isSourceGradeScaleUpdated = false;
+        if (bonus.getSourceGradingScale() != null && !oldBonus.getSourceGradingScale().getId().equals(bonus.getSourceGradingScale().getId())) {
+            var sourceFromDb = gradingScaleRepository.findById(bonus.getSourceGradingScale().getId()).orElseThrow();
+            bonus.setSourceGradingScale(sourceFromDb);
+            checkIsAtLeastInstructorForGradingScaleCourse(sourceFromDb);
+            isSourceGradeScaleUpdated = true;
+        }
+
+        bonusToGradingScale.addBonusFrom(bonus);
+        bonusToGradingScale.setBonusStrategy(bonus.getBonusStrategy());
+        gradingScaleRepository.save(bonusToGradingScale);
+        Bonus savedBonus = bonusService.saveBonus(bonus, isSourceGradeScaleUpdated);
+
+        filterBonusForResponse(savedBonus);
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, "")).body(savedBonus);
+    }
+
+    private void checkIsAtLeastInstructorForGradingScaleCourse(GradingScale gradingScale) {
+        Course sourceCourse = gradingScale.getCourseViaExamOrDirectly();
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, sourceCourse, null);
+    }
+
+    /**
+     * DELETE /courses/{courseId}/bonus : Delete bonus source for course
+     *
+     * @param bonusId the id of the bonus source to delete
+     * @return ResponseEntity with status 200 (Ok) if the bonus source is successfully deleted and 400 (Bad request) otherwise
+     */
+    @DeleteMapping("/bonus/{bonusId}")
```

Comment:
wrong URL

Reply:
Url is standardized by adding `courseId` and `examId`.